### PR TITLE
[docs] Fix M4C docs and move them to website for new api

### PR DESCRIPTION
**Pull request by apsdehal**
- Move docs to projects url in website
- Fix docs according to new API

Test Plan:
Deployed the website locally

### CODE REVIEW

File: projects/m4c/configs/ocrvqa/defaults.yaml [hunk 1]
```diff
@@ -87,7 +87,7 @@ training:
     warmup_iterations: 1000
     max_iterations: 48000
```
Comment (ronghanghu) [inline projects/m4c/configs/ocrvqa/defaults.yaml:88]: I think this should be updated to `max_updates` to match the latest config

File: projects/m4c/configs/stvqa/defaults.yaml [hunk 2]
```diff
@@ -88,7 +88,7 @@ training:
     warmup_iterations: 1000
     max_iterations: 24000
```
Comment (ronghanghu) [inline projects/m4c/configs/stvqa/defaults.yaml:89]: same here

File: projects/m4c/configs/textvqa/defaults.yaml [hunk 3]
```diff
@@ -87,7 +87,7 @@ training:
     warmup_iterations: 1000
     max_iterations: 24000
```
Comment (ronghanghu) [inline projects/m4c/configs/textvqa/defaults.yaml:88]: and here

File: website/docs/projects/m4c.md [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+Project Page: http://ronghanghu.com/m4c
+
+## Installation
+
+Clone MMF, and build it with the following command.
+
+```
+cd ~/mmf
+python setup.py build develop
+```
```
Comment (ronghanghu) [inline website/docs/projects/m4c.md:29]: Since MMF is now fully merged into the Master branch, we can just ask users to follow the standard MMF installation procedure. So we can change the 6 lines above into:
```
Install MMF following the [installation guide](https://mmf.sh/docs/getting_started/installation/).
```

File: website/docs/projects/m4c.md [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+Please follow the [Pythia documentation](https://mmf.sh/docs/getting_started/quickstart#training) for the training and evaluation of the M4C model on each dataset.
+
+For example:
+
+1. to train the M4C model on the TextVQA training set:
+
+```
+mmf_run dataset=textvqa \
+  model=m4c \
+  config=projects/m4c/textvqa/defaults.yaml \
```
Comment (ronghanghu) [inline website/docs/projects/m4c.md:70]: The config file should be `projects/m4c/configs/textvqa/defaults.yaml` instead.

File: website/docs/projects/m4c.md [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+  training.save_dir=./save/m4c
+```
+
+(Replace `textvqa` with other datasets and `projects/m4c/textvqa/defaults.yaml` with other config files to train with other datasets and configurations. See the table above. You can also specify a different path to `training.save_dir` to save to a location you prefer.)
```
Comment (ronghanghu) [inline website/docs/projects/m4c.md:74]: same here

File: website/docs/projects/m4c.md [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+
+For example:
+
+1. to train the M4C model on the TextVQA training set:
+
+```
+mmf_run dataset=textvqa \
+  model=m4c \
+  config=projects/m4c/textvqa/defaults.yaml \
+  training.save_dir=./save/m4c
```
Comment (ronghanghu) [inline website/docs/projects/m4c.md:71]: This should be `env.save_dir` instead.

File: website/docs/projects/m4c.md [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+
+(Replace `textvqa` with other datasets and `projects/m4c/textvqa/defaults.yaml` with other config files to train with other datasets and configurations. See the table above. You can also specify a different path to `training.save_dir` to save to a location you prefer.)
+
+2. To evaluate the pretrained M4C model locally on the a TextVQA's validation set (assuming that the pretrained model that you are evaluating is `m4c.textvqa.with_stvqa`):
+
+```
+mmf_run dataset=textvqa \
+  model=m4c \
+  config=projects/m4c/configs/textvqa/defaults.yaml \
+  training.save_dir=save/m4c \
```
Comment (ronghanghu) [inline website/docs/projects/m4c.md:82]: same here (env.save_dir).

File: website/docs/projects/m4c.md [hunk 9]
```diff
@@ -0,0 +1,118 @@
...
+
+:::
+
+3. to generate the EvalAI prediction files for the TextVQA test set (assuming you are evaluating the pretrained model `m4c.textvqa.with_stvqa`):
+
+```
+mmf_predict dataset=textvqa \
+  model=m4c \
+  config=projects/m4c/configs/textvqa/defaults.yaml \
+  training.save_dir=save/m4c \
```
Comment (ronghanghu) [inline website/docs/projects/m4c.md:107]: same here (`env.save_dir`).

File: website/docs/projects/m4c.md [hunk 10]
```diff
@@ -0,0 +1,118 @@
...
+| --- | --- | --- | --- | --- |
+| TextVQA (`textvqa`) | `textvqa/join_with_stvqa.yaml` | `m4c.textvqa.with_stvqa` | val accuracy - 40.55%; test accuracy - 40.46% | Rosetta-en OCRs; ST-VQA as additional data |
+| TextVQA (`textvqa`) | `textvqa/defaults.yaml` | `m4c.textvqa.alone` | val accuracy - 39.40%; test accuracy - 39.01% | Rosetta-en OCRs |
+| TextVQA (`textvqa`) | `textvqa/ocr_ml.yaml` | m4c.textvqa.ocr_ml | val accuracy - 37.06% | Rosetta-ml OCRs |
+| ST-VQA (`stvqa`) | `stvqa/defaults.yaml` | `m4c.stvqa.defaults` | val ANLS - 0.472 (accuracy - 38.05%); test ANLS - 0.462 | Rosetta-en OCRs |
+| OCR-VQA (`ocrvqa`) | `ocrvqa/defaults.yaml` | `m4c.ocrvqa.defaults` | val accuracy - 63.52%; test accuracy - 63.87% | Rosetta-en OCRs |
+
+## Training and Evaluation
+
+Please follow the [Pythia documentation](https://mmf.sh/docs/getting_started/quickstart#training) for the training and evaluation of the M4C model on each dataset.
```
Comment (ronghanghu) [inline website/docs/projects/m4c.md:61]: Let's replace Pythia with MMF :)

File: projects/m4c/README.md [hunk 11]
```diff
@@ -1,108 +1,3 @@
...
-* R. Hu, A. Singh, T. Darrell, M. Rohrbach, *Iterative Answer Prediction with Pointer-Augmented Multimodal Transformers for TextVQA*. in CVPR, 2020 ([PDF](https://arxiv.org/pdf/1911.06258.pdf))
-```
-@inproceedings{hu2020iterative,
-  title={Iterative Answer Prediction with Pointer-Augmented Multimodal Transformers for TextVQA},
-  author={Hu, Ronghang and Singh, Amanpreet and Darrell, Trevor and Rohrbach, Marcus},
-  booktitle={Proceedings of the IEEE Conference on Computer Vision and Pattern Recognition},
-  year={2020}
-}
-```
-
```
Comment (vedanuj) [inline projects/m4c/README.md:14]: Let's keep the citation here atleast(or if you want it in docs, keep a copy over here as well) for consistency with other projects.

File: website/docs/projects/m4c.md [hunk 12]
```diff
@@ -0,0 +1,118 @@
...
+
+For example:
+
+1. to train the M4C model on the TextVQA training set:
+
+```
```
Comment (vedanuj) [inline website/docs/projects/m4c.md:67]: Let's make these bash code blocks. Same for the other commands below.

**Comment by ronghanghu** [comment]
Thanks for the PR! Looks to me overall. I made a few minor changes.

It seems that I cannot directly update (or add commits to) this PR. GitHub rejects my push, likely because I'm not authorized to push to this repo. I amended the changes in the [internal branch](https://github.com/fairinternal/mmf-internal/tree/m4c_readme_update).

**Comment by vedanuj** [comment]
Fixes #331
